model-optimizer/extensions/ops/slice_like.py

```diff
@@ -52,3 +52,15 @@ def infer(node):
             new_shape = np.copy(shape_like)
 
         node.out_port(0).data.set_shape(new_shape)
+        if node.in_port(0).get_connection().data.get_value() is not None and \
```

Comment:
I'm not sure if we really would like to have this dependence on keep_shape_ops key. Could you explain why we do that? Is it because of slice_like nature (using second input to take shape of it)?

Reply:
@jane-intel It is necessary for reshape-ability
I guess, value propagation here is the only way to fix the problem with IR reader. From the other side, it breaks reshape-ability, because there will be a constant node after partial infer.
This dependence allows us to save slice_like op in graph and then replace it with StridedSlice

Reply:
Lets revert the keep shape ops logic here and make clean_up change by adding SliceLike there. And we should make `mark_undead_nodes` function to filter by op too.